### PR TITLE
RavenDB-12269 Fixing handling of index replacement

### PR DESCRIPTION
**Pull request by arekpalinski**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -554,6 +551,20 @@ internal ExitWriteLock DrainRunningQueries()
             }
         }
 
+        private void InitializeComponentsUsingEnvironment(DocumentDatabase documentDatabase, StorageEnvironment environment)
+        {
+            _contextPool?.Dispose();
+            _contextPool = new TransactionContextPool(environment);
+            
+            _indexStorage = new IndexStorage(this, _contextPool, documentDatabase);
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Index.cs:559]: You aren't calling `_indexStorage ?.Dispose()` intentionally?

File: test/SlowTests/Issues/RavenDB_12269.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+                db.IndexStore.StartIndexing();
+
+                WaitForIndexing(store);
+
+                // let's try to force calling storageEnvironment.Cleanup() inside ExecuteIndexing method
+                replacementIndex.LowMemory();
+                replacementIndex._indexStorage.Environment().LogsApplied();
+
+                Thread.Sleep(500);
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB_12269.cs]: This is recipe for test failures. Can you add an event, instead? Or do active wiating?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Index.cs:559]: It is not disposable, it just wraps _environment

File: test/SlowTests/Issues/RavenDB_12269.cs (hunk 2 above)
Comment (arekpalinski) [reply test/SlowTests/Issues/RavenDB_12269.cs]: Actually we don't need that. The below query will assert that that index state isn't Error. Fixed in https://github.com/ravendb/ravendb/pull/7983/commits/3094df574da4e8f74005637ab4ebae1211f8c631

**Review by ppekrol — APPROVED** [review]

**Pull request merged**